keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -390,13 +401,24 @@ const {
       // ServiceExportExported means that the service referenced by this
       // service export has been synced to all clusters in the supercluster
       ServiceExportExported ServiceExportConditionType = "Exported"
+      // ServiceExportInvalid means that the service marked for export has an
+      // unexportable service type (ExternalName)
+      ServiceExportInvalid ServiceExportConditionType = "InvalidServiceType"
+      // ServiceExportHeadless describes the headlessness of the supercluster
+      // service. Only required to be set if at least one cluster (including
+      // the local one) has a matching headless service marked for export.
+      // "True" when the corresponding ServiceImport is headless. "False" if
+      // at least one matching service is not headless.
+      // If any exported services disagree, cluster names and respective
+      // headlessness should be listed in the condition message.
+      ServiceExportHeadless ServiceExportConditionType = "Headless"
```

Comment:
Why are we back-populating this into Export?  Shouldn't the data in the Export be about the local export, only?

Reply:
Discussed in the SIG meeting, but repeating here for future readers.

This condition allows users to understand if they've created a headlessness conflict while looking only at the status of the most recently deployed ServiceExport. Since we're saying that a multi-cluster service is headless only if all of its backing services are headless, any non-headless export can revert a multi-cluster service to SuperclusterIP. We don't have a way to tell which is the expected state if there is any disagreement among services, so we need to surface derived headlessness and some descriptive message explaining the conflict on each export